build.go

```diff
@@ -57,50 +51,38 @@ func Build(
 
 		goCacheLayer.Cache = true
 
-		checksum, err := checksumCalculator.Sum(context.WorkingDir)
+		// Parse the BuildConfiguration from the environment again since a prior
+		// step may have augmented the configuration.
+		configuration, err := parser.Parse(context.WorkingDir)
+		if err != nil {
+			return packit.BuildResult{}, packit.Fail.WithMessage("failed to parse build configuration: %w", err)
+		}
+
+		goPath, path, err := pathManager.Setup(context.WorkingDir, configuration.ImportPath)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		command, err := buildProcess.Execute(GoBuildConfiguration{
+			Workspace: path,
+			Output:    filepath.Join(targetsLayer.Path, "bin"),
```

Comment:
why build into a layer and not just into the workspace (or a temp dir, then move into the workspace)?

Reply:
I think that the biggest reason to build it into a layer is that the binaries that are produced are put into the `bin` directory of the layer which puts them directly on the path. If we were to want to keep the same behavior of having built binaries on the path we would have to put the WorkspaceDir on the path which would require the construction of a layer to store the environment variable.

If I were also to speak from a understandability angle, I personally find it easier to reason that the artifacts where built somewhere and then the source was deleted vs the artifacts were built somewhere, the source was deleted, and then the artifacts where shuffled back.

Reply:
👍  that makes sense to me. I was thinking more about consistency across buildpacks. In dotnet publish, we pop built artifacts right into the working directory, and avoid the layer thing entirely. We get away with this by setting start commands like `./my-app-binary`, rather than relying on the binary being on the path. I guess the difference is that `dotnet-publish` and `dotnet-execute` are separate buildpacks, and building into a layer in `publish` then requires some inter-buildpack communication.